change-streams/reacting-to-database-changes-with-mongodb-change-streams-and-go.md

```diff
@@ -0,0 +1,204 @@
+
+## Creating a Real-Time MongoDB Change Stream with Golang
+
+While there are many possible use-cases for change streams, we're going to continue with the example that we've been using throughout the scope of this getting started series. We're going to continue working with podcast show and podcast episode data.
+
+Let's assume we have the following code to start:
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func main() {
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(os.Getenv("ATLAS_URI")))
+	if err != nil {
+		panic(err)
+	}
+	defer client.Disconnect(context.TODO())
+
+	database := client.Database("quickstart")
+	episodesCollection := database.Collection("episodes")
+}
+```
+
+The above code is a very basic connection to a MongoDB cluster, something that we explored in the [How to Get Connected to Your MongoDB Cluster with Go](https://www.mongodb.com/blog/post/quick-start-golang--mongodb--starting-and-setup), tutorial.
+
+To watch for changes, we can do something like the following:
+
+```go
+episodesStream, err := episodesCollection.Watch(context.TODO(), mongo.Pipeline{})
+if err != nil {
+    panic(err)
+}
+```
+
+The above code will watch for any and all changes to documents within the `episodes` collection. The result is a cursor that we can iterate over indefinitely for data as it comes in.
+
+We can iterate over the curser and make sense of our data using the following code:
+
+```go
+episodesStream, err := episodesCollection.Watch(context.TODO(), mongo.Pipeline{})
+if err != nil {
+    panic(err)
+}
+
+defer episodesStream.Close(context.TODO())
+
+for episodesStream.Next(context.TODO()) {
+    var data bson.M
+    if err := episodesStream.Decode(&data); err != nil {
+        panic(err)
+    }
+    fmt.Printf("%v\n", data)
+}
+```
+
+If data were to come in, it might look something like the following:
+
+```
+map[_id:map[_data:825E4EFCB9000000012B022C0100296E5A1004D960EAE47DBE4DC8AC61034AE145240146645F696400645E3B38511C9D4400004117E80004] clusterTime:{1582234809 1} documentKey:map[_id:ObjectID("5e3b38511c9d
+4400004117e8")] fullDocument:map[_id:ObjectID("5e3b38511c9d4400004117e8") description:The second episode duration:30 podcast:ObjectID("5e3b37e51c9d4400004117e6") title:Episode #3] ns:map[coll:episodes 
+db:quickstart] operationType:replace]
+```
+
+In the above example, I've done a `Replace` on a particular document in the collection. In addition to information about the data, I also receive the full document that includes the change. The results will vary depending on the `operationType` that takes place.
+
+While the code that we used would work fine, it is currently a blocking operation. If we wanted to watch for changes and continue to do other things, we'd want to use a goroutine for iterating over our change stream cursor.
```

Comment:
I would maybe add a link to goroutine that explains what it is on the Golang homepage or go by example.